security/auto-host-endpoints.md

```diff
@@ -0,0 +1,181 @@
+For named host endpoints, {{site.prodname}} blocks traffic only to/from interfaces that it’s been explicitly told about in network policy. Traffic to/from other interfaces is ignored.
+For wildcard host endpoints, {{site.prodname}} blocks traffic to/from _all_ non-workload interfaces on the host (except for traffic allowed by failsafe rules).
+
+However, profiles can be used in conjunction with host endpoints to modify default behavior of external traffic to/from the host in the absence of network policy.
+{{site.prodname}} provides a default profile resource named `projectcalico-allow-all` that consists of allow-all ingress and egress rules.
+Host endpoints with the `projectcalico-allow-all` profile attached will have "allow-all" semantics instead of "deny-all" in the absence of policy.
+
+> Auto host endpoints have the `projectcalico-allow-all` profile attached and thus they allow all traffic in the absence of policy.
+{: .alert .alert-info}
+
+### Before you begin...
+
+Have a running {{site.prodname}} cluster with calicoctl installed.
+
+### How to
+
+- [Enable automatic host endpoints](#enable-automatic-host-endpoints)
+- [Restrict host egress to whitelisted IPs](#restrict-host-egress-to-whitelisted-ips)
+
+#### Enable automatic host endpoints
+
+In order to enable automatic host endpoints, we need to edit the default KubeControllersConfiguration instance.
+We will be setting `spec.controllers.node.hostEndpoint.autoCreate` to `true`.
+
+```bash
+calicoctl patch kubecontrollersconfiguration default --patch='{"spec": {"controllers": {"node": {"hostEndpoint": {"autoCreate": "Enabled"}}}}}'
+```
+
+If the apply was successful, we should see host endpoints created for each of your cluster's nodes:
+
+```bash
+calicoctl get heps -owide
+```
+
+The output may look similar to this:
+
+```
+$ calicoctl get heps -owide
+NAME                                                    NODE                                           INTERFACE   IPS                              PROFILES
+ip-172-16-101-147.us-west-2.compute.internal-auto-hep   ip-172-16-101-147.us-west-2.compute.internal   *           172.16.101.147,192.168.228.128
+ip-172-16-101-54.us-west-2.compute.internal-auto-hep    ip-172-16-101-54.us-west-2.compute.internal    *           172.16.101.54,192.168.107.128
+ip-172-16-101-79.us-west-2.compute.internal-auto-hep    ip-172-16-101-79.us-west-2.compute.internal    *           172.16.101.79,192.168.91.64
+ip-172-16-101-9.us-west-2.compute.internal-auto-hep     ip-172-16-101-9.us-west-2.compute.internal     *           172.16.101.9,192.168.71.192
+ip-172-16-102-63.us-west-2.compute.internal-auto-hep    ip-172-16-102-63.us-west-2.compute.internal    *           172.16.102.63,192.168.108.192
+```
+
+#### Apply network policy to automatic host endpoints
+
+Automatic host endpoints share the common label `projectcalico.org/created-by: calico-kube-controllers`.
+To write policy that targets all Kubernetes nodes, use that label:
```

Comment:
Not sure I like this. Seems like a semantic mismatch. kube-controllers creates lots of things, and may create more things in the future - i.e. created by kube-controllers doens't sound like a guaranteed synonym for "host endpoints for all kubernetes nodes".  In reality today it might be, but I'm hesitant to encourage this kind of usage.

Maybe a better best practice would be to use namespaceSelector:global(), which would select all host-endpoints (because workload endpoints are always namespaced).

Reply:
I see what you mean. Hmm... unfortunately, `global()` is only in enteprise currently.

Reply:
In the end, I didn't use an "all nodes" label. There is an open PR to port `global()`.